### PR TITLE
Add `setup.py develop` to documentation

### PR DESCRIPTION
**Pull request by hypothesist**
Add `python setup.py develop` to the
documentation, explaining what it does and
its uses.

Fixes https://github.com/coala-analyzer/coala/issues/1390

### CODE REVIEW

File: docs/Users/Install.rst [hunk 1]
```diff
@@ -126,6 +126,16 @@ You will have coala installed into your python scripts directory. On an
...
+You may also install a development version of coala to test and make changes
+easily. To do this run:
+
+::
+
+    $ python3 setup.py develop
+
+This essentially lets you install coala in a way that allows you to make
+changes to the code and see the changes take effect immediately.
+
```
Comment (sudheesh001) [inline docs/Users/Install.rst]: http://pythonhosted.org//setuptools/setuptools.html#development-mode

I think I would also mention the fact about the .egg and recommend those willing to contribute to use `develop` and the others to use `install`

**Comment by Makman2** [comment]
ack d435d49

**Comment by Makman2** [comment]
@hypothesist a little rebase and your PR is ready for merge :)

**Comment by Makman2** [comment]
@hypothesist please don't click this (damn) `Merge pull request` button on GitHub :) We use fast-forward merges so we get rid of these merge-commits since they are just cluttering commit history. You need to invoke locally `git rebase upstream/master` (before you need to fetch the latest changes via `git fetch upstream/master`) and do a `git push --force`. If you can't push force onto your branch because it's protected, you have two options:
1. deactivate temporarily the branch protection on your master and do `git push --force` again
2. move everything to a new (non-protected) branch and open a new pull request

In this case I would suggest doing 1. Next time please open a separate branch even in your fork and work on that instead on master :)

**Comment by sils1297** [comment]
@hypothesist need help? Want to drop this?

**Comment by sudheesh001** [comment]
@hypothesist Are you still working on this one ? :smile: You could always reach out to us at https://gitter.im/coala-analyzer/coala in case there's an issue.

**Comment by rnikhil275** [comment]
I made some little changes. Please review it [here](https://github.com/coala-analyzer/coala/pull/1628)

**Pull request closed without merging**